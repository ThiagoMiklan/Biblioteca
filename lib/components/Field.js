"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _Icon = _interopRequireDefault(require("./Icon"));

var _Control = _interopRequireDefault(require("./Control"));

var _Input = _interopRequireDefault(require("./Input"));

var _type_validations = require("../tools/type_validations.js");

var props_obj = {
  label: "string",
  control_definition: "string",
  input_definition: "string",
  icon_definition: "string",
  icon_name: "string",
  onChange: "function",
  placeholder: "string",
  type: "string",
  value: "string"
};

var Field = function Field(props) {
  (0, _type_validations.validate)(props, props_obj, "Field");
  var help_definition = props.help_definition == undefined ? "help" : "help " + props.help_definition;
  return /*#__PURE__*/_react.default.createElement("div", {
    className: "field"
  }, /*#__PURE__*/_react.default.createElement("label", {
    className: "label"
  }, props.label), /*#__PURE__*/_react.default.createElement(_Control.default, {
    definition: props.control_definition == undefined ? "has-icons-left" : props.control_definition
  }, /*#__PURE__*/_react.default.createElement(_Input.default, {
    definition: props.input_definition,
    onChangeEvent: props.onChange,
    placeholder: props.placeholder,
    type: props.type,
    value: props.value,
    custom: props.custom
  }), assembleIcon(props), props.help_value != undefined && /*#__PURE__*/_react.default.createElement("p", {
    className: help_definition
  }, props.help_value)));
};

function assembleIcon(props) {
  var code = /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null);

  if (props.control_definition != undefined) {
    var right = props.control_definition.includes("has-icons-right");

    if (right == true) {
      var definition = props.icon_definition == undefined ? "" : props.icon_definition;
      var icon_def = definition + " is-right";
      code = /*#__PURE__*/_react.default.createElement(_Icon.default, {
        definition: icon_def,
        icon_name: props.icon_name
      });
    } else {
      code = /*#__PURE__*/_react.default.createElement(_Icon.default, {
        definition: props.icon_definition,
        icon_name: props.icon_name
      });
    }
  } else {
    code = /*#__PURE__*/_react.default.createElement(_Icon.default, {
      definition: props.icon_definition,
      icon_name: props.icon_name
    });
  }

  return code;
}

var _default = Field;
exports.default = _default;