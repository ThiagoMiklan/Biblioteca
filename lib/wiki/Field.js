"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _react = _interopRequireWildcard(require("react"));

var _reactDom = _interopRequireDefault(require("react-dom"));

var _Field = _interopRequireDefault(require("./components/Field"));

require("bulma/css/bulma.css");

var _Columns = _interopRequireDefault(require("./components/Columns"));

var _Column = _interopRequireDefault(require("./components/Column"));

var custom = {
  draggable: true
};

_reactDom.default.render( /*#__PURE__*/_react.default.createElement("div", {
  className: "container is-fluid"
}, /*#__PURE__*/_react.default.createElement(_Field.default, {
  label: "Example",
  input_definition: "is-primary",
  icon_definition: "is-primary",
  icon_name: "fas fa-user",
  placeholder: "testing",
  control_definition: "has-icons-right"
}), /*#__PURE__*/_react.default.createElement(_Field.default, {
  label: "Example",
  input_definition: "is-info",
  icon_definition: "is-info",
  icon_name: "fas fa-user",
  placeholder: "testing",
  control_definition: "has-icons-right"
}), /*#__PURE__*/_react.default.createElement(_Field.default, {
  label: "Example",
  input_definition: "is-success",
  icon_definition: "is-success",
  icon_name: "fas fa-user",
  placeholder: "testing",
  control_definition: "has-icons-right"
}), /*#__PURE__*/_react.default.createElement(_Field.default, {
  label: "Example",
  input_definition: "is-danger",
  icon_definition: "is-danger",
  icon_name: "fas fa-user",
  placeholder: "testing",
  control_definition: "has-icons-right"
}), /*#__PURE__*/_react.default.createElement(_Field.default, {
  label: "Example",
  input_definition: "is-warning",
  icon_definition: "is-warning",
  placeholder: "testing",
  icon_name: "fas fa-user",
  control_definition: "has-icons-right"
}), /*#__PURE__*/_react.default.createElement(_Field.default, {
  label: "Example",
  input_definition: "is-small",
  icon_definition: "is-warning",
  icon_name: "fas fa-user",
  control_definition: "has-icons-right"
}), /*#__PURE__*/_react.default.createElement(_Field.default, {
  label: "Example",
  input_definition: "is-medium",
  icon_definition: "is-warning",
  icon_name: "fas fa-user",
  control_definition: "has-icons-right"
}), /*#__PURE__*/_react.default.createElement(_Field.default, {
  label: "Example",
  input_definition: "is-large",
  icon_definition: "is-warning",
  icon_name: "fas fa-user",
  control_definition: "has-icons-right"
}), /*#__PURE__*/_react.default.createElement(_Field.default, {
  label: "Example",
  input_definition: "is-small is-rounded",
  control_definition: "has-icons-right"
}), /*#__PURE__*/_react.default.createElement(_Field.default, {
  label: "Example",
  input_definition: "is-medium is-rounded",
  control_definition: "has-icons-right"
}), /*#__PURE__*/_react.default.createElement(_Field.default, {
  label: "Example",
  input_definition: "is-large is-rounded",
  control_definition: "has-icons-right"
}), /*#__PURE__*/_react.default.createElement(_Columns.default, null, /*#__PURE__*/_react.default.createElement(_Column.default, {
  definition: "is-half"
}, /*#__PURE__*/_react.default.createElement(_Field.default, {
  label: "Example Help",
  input_definition: "is-large is-rounded",
  control_definition: "has-icons-right",
  help_value: "This is a help message",
  help_definition: "is-success"
}))), /*#__PURE__*/_react.default.createElement("br", null)), document.getElementById('root'));