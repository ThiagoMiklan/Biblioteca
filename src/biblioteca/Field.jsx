import React from 'react';
import PropTypes from 'prop-types'
import Icon from './Icon';
import Control from './Control';
import Input from './Input';

// O presente componente foi criado como maneira de facilitar
// o uso do componente Input,Control,Label e Icon,
// Buscando fornecer um componente visual que mostra um campo de texto
// comum, com um rótulo e um espaço para digitação, mesclando os quatro componentes
// antes citado
// Obs: Principal motivação, o excesso de código, assim como sua repetição
// Eventos : onChange do componente Input

// Para fim  de organização. Considera-se esse um componente misto.

const Field = (props) => {

    return <>
        <label className="label">{props.label}</label>
        <Control definition={props.control_definition}>
            <Input definition={props.input_definition} onChangeEvent={props.onChange} placeholder={props.placeholder} type={props.type} value={props.value}></Input>
            <Icon definition={props.icon_definition} icon_name={props.icon_name}></Icon>
        </Control>
        </>

}


Field.propTypes = {
    /*
    Nome do rótulo que irá aparecer sobre o 
    input
    */
    label: PropTypes.string,
    /*
    definition do componente control,
    qual característica pré-defininada do 
    componente, é desejada que apareça no 
    componente Field.
    */
    control_definition: PropTypes.string,
    /*
    definition do componente input,
    qual característica pré-defininada do 
    componente, é desejada que apareça no 
    componente Field.
    */
    input_definition: PropTypes.string,
    /*
    definition do componente control,
    qual característica pré-defininada do 
    componente, é desejada que apareça no 
    componente Field.
    Essa propriedade é importante pois
    caso deseje-se colocar o icon no lado 
    direito e/ou esquerdo, deve se utilizar essa propriedade
    e informar "right" ou "left"
    */
    icon_definition: PropTypes.string,
    /*
    Referente ao nome do ícone segundo
    o Font Awesome
    */
    icon_name: PropTypes.string,
    /*
    Função onChange do componente Input
    */
    onChange: PropTypes.func,
    // Placeholder Input
    placeholder: PropTypes.string,
    // Type Input
    type: PropTypes.string,
    // Valor opcional de valor do input do field
    value: PropTypes.string
}


export default Field;