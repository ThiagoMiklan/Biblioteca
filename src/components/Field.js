// @flow
import React from 'react';
import Icon from './Icon';
import Control from './Control';
import Input from './Input';
import { validate } from '../tools/type_validations.js';

type Props = {
    label?: string,
    control_definition?: string,
    input_definition?: string,
    icon_definition?: any,
    icon_name?: string,
    onChange?: () => void,
    placeholder?: string,
    type?: string,
    value?: string,
    custom?: Object,
    help_definition?: string,
    help_value?: string
}

const props_obj ={
    label: "string",
    control_definition: "string",
    input_definition: "string",
    icon_definition: "string",
    icon_name: "string",
    onChange: "function",
    placeholder: "string",
    type: "string",
    value: "string"
}


const Field = (props: Props) => {
    validate(props,props_obj, "Field");
    let help_definition = (props.help_definition == undefined) ? "help": "help "+props.help_definition;
    return <div className="field">
        <label className="label">{props.label}</label>
        <Control definition={props.control_definition == undefined ? "has-icons-left" : props.control_definition}>
            <Input  definition={props.input_definition} 
                    onChangeEvent={props.onChange}
                    placeholder={props.placeholder} 
                    type={props.type} 
                    value={props.value}
                    custom={props.custom}
                    ></Input>
            {assembleIcon(props)}
            {props.help_value != undefined &&
            <p className={help_definition}>{props.help_value}</p>
            }
        </Control>
    </div>
}

function assembleIcon(props) {
    var code = <></>;
    if (props.control_definition != undefined) {
        var right = props.control_definition.includes("has-icons-right");
        if (right == true) {
            var definition = (props.icon_definition == undefined) ? "" : props.icon_definition;
            var icon_def = definition + " is-right";
            code = <Icon definition={icon_def} icon_name={props.icon_name}></Icon>;
        } else {
            code = <Icon definition={props.icon_definition} icon_name={props.icon_name}></Icon>;
        }
    } else {
        code = <Icon definition={props.icon_definition} icon_name={props.icon_name}></Icon>;
    }

    return code;
}


export default Field;