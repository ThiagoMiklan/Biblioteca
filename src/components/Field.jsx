// @flow
import React from 'react';
import Icon from './Icon';
import Control from './Control';
import Input from './Input';
import { validate } from '../tools/type_validations.js';

// Custom

type Props = {
    label?: string,
    control_definition?: string,
    input_definition?: string,
    icon_definition?: any,
    icon_name?: string,
    onChange?: () => void,
    placeholder?: string,
    type?: string,
    value?: string,
    custom?: Object
}

const props_obj ={
    label: "string",
    control_definition: "string",
    input_definition: "string",
    icon_definition: "string",
    icon_name: "string",
    onChange: "function",
    placeholder: "string",
    type: "string",
    value: "string"
}


const Field = (props: Props) => {
    validate(props,props_obj, "Field");
    return <div className="field">
        <label className="label">{props.label}</label>
        <Control definition={props.control_definition == undefined ? "has-icons-left" : props.control_definition}>
            <Input  definition={props.input_definition} 
                    onChangeEvent={props.onChange}
                    placeholder={props.placeholder} 
                    type={props.type} 
                    value={props.value}
                    custom={props.custom}
                    ></Input>
            {assembleIcon(props)}
        </Control>
    </div>
}

function assembleIcon(props) {
    var code = <></>;
    if (props.control_definition != undefined) {
        var right = props.control_definition.includes("has-icons-right");
        if (right == true) {
            var definition = (props.icon_definition == undefined) ? "" : props.icon_definition;
            var icon_def = definition + " is-right";
            code = <Icon definition={icon_def} icon_name={props.icon_name}></Icon>;
        } else {
            code = <Icon definition={props.icon_definition} icon_name={props.icon_name}></Icon>;
        }
    } else {
        code = <Icon definition={props.icon_definition} icon_name={props.icon_name}></Icon>;
    }

    return code;
}


export default Field;