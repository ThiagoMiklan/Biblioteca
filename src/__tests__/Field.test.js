import React from 'react';
import Field from '../biblioteca/Field'
import renderer from 'react-test-renderer';

test('fields colors', () => {
    const colors = renderer.create(
        <>
            <Field label="Example"
                input_definition="primary"
                icon_definition="primary"
                icon_name="fas fa-user"
                placeholder="testing"
                control_definition="icons-right">
            </Field>
            <Field label="Example"
                input_definition="info"
                icon_definition="info"
                icon_name="fas fa-user"
                placeholder="testing"
                control_definition="icons-right">
            </Field>
            <Field label="Example"
                input_definition="success"
                icon_definition="success"
                icon_name="fas fa-user"
                placeholder="testing"
                control_definition="icons-right">
            </Field>
            <Field label="Example"
                input_definition="danger"
                icon_definition="danger"
                icon_name="fas fa-user"
                placeholder="testing"
                control_definition="icons-right">
            </Field>
            <Field label="Example"
                input_definition="warning"
                icon_definition="warning"
                placeholder="testing"
                icon_name="fas fa-user"
                control_definition="icons-right">
            </Field>
        </>
    ).toJSON();

    expect(colors).toMatchSnapshot();
})

test('sizes',()=>{

    const sizes = renderer.create(
        <>
        <Field label="Example"
               input_definition="small"
               icon_definition="warning"
               icon_name="fas fa-user"
               control_definition="icons-right">
        </Field>

        <Field label="Example"
               input_definition="medium"
               icon_definition="warning"
               icon_name="fas fa-user"
               control_definition="icons-right">
        </Field>

        <Field label="Example"
               input_definition="large"
               icon_definition="warning"
               icon_name="fas fa-user"
               control_definition="icons-right">
        </Field>
        </>
    ).toJSON();

    expect(sizes).toMatchSnapshot();
})

test('sizes rounded',()=>{
    const rounded = renderer.create(
        <>
        <Field label="Example"
               input_definition={"small rounded"}
               control_definition="icons-right">
        </Field>

        <Field label="Example"
               input_definition={"medium rounded"}
               control_definition="icons-right">
        </Field>

        <Field label="Example"
               input_definition={"large rounded"}
               control_definition="icons-right">
        </Field>
        </>
    ).toJSON();

    expect(rounded).toMatchSnapshot();
})