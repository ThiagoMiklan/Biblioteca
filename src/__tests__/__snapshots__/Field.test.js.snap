// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fields colors 1`] = `
Array [
  <div
    className="field"
  >
    <label
      className="label"
    >
      Example
    </label>
    <div
      className="control has-icons-right"
    >
      <div
        className="control"
      >
        <input
          className="input is-primary"
          placeholder="testing"
        />
      </div>
      <span
        className="icon is-primary is-right"
      >
        <i
          className="fas fa-user"
        />
      </span>
      <span />
    </div>
  </div>,
  <div
    className="field"
  >
    <label
      className="label"
    >
      Example
    </label>
    <div
      className="control has-icons-right"
    >
      <div
        className="control"
      >
        <input
          className="input is-info"
          placeholder="testing"
        />
      </div>
      <span
        className="icon is-info is-right"
      >
        <i
          className="fas fa-user"
        />
      </span>
      <span />
    </div>
  </div>,
  <div
    className="field"
  >
    <label
      className="label"
    >
      Example
    </label>
    <div
      className="control has-icons-right"
    >
      <div
        className="control"
      >
        <input
          className="input is-success"
          placeholder="testing"
        />
      </div>
      <span
        className="icon is-success is-right"
      >
        <i
          className="fas fa-user"
        />
      </span>
      <span />
    </div>
  </div>,
  <div
    className="field"
  >
    <label
      className="label"
    >
      Example
    </label>
    <div
      className="control has-icons-right"
    >
      <div
        className="control"
      >
        <input
          className="input is-danger"
          placeholder="testing"
        />
      </div>
      <span
        className="icon is-danger is-right"
      >
        <i
          className="fas fa-user"
        />
      </span>
      <span />
    </div>
  </div>,
  <div
    className="field"
  >
    <label
      className="label"
    >
      Example
    </label>
    <div
      className="control has-icons-right"
    >
      <div
        className="control"
      >
        <input
          className="input is-warning"
          placeholder="testing"
        />
      </div>
      <span
        className="icon is-warning is-right"
      >
        <i
          className="fas fa-user"
        />
      </span>
      <span />
    </div>
  </div>,
]
`;

exports[`sizes 1`] = `
Array [
  <div
    className="field"
  >
    <label
      className="label"
    >
      Example
    </label>
    <div
      className="control has-icons-right"
    >
      <div
        className="control"
      >
        <input
          className="input is-small"
        />
      </div>
      <span
        className="icon is-warning is-right"
      >
        <i
          className="fas fa-user"
        />
      </span>
      <span />
    </div>
  </div>,
  <div
    className="field"
  >
    <label
      className="label"
    >
      Example
    </label>
    <div
      className="control has-icons-right"
    >
      <div
        className="control"
      >
        <input
          className="input is-medium"
        />
      </div>
      <span
        className="icon is-warning is-right"
      >
        <i
          className="fas fa-user"
        />
      </span>
      <span />
    </div>
  </div>,
  <div
    className="field"
  >
    <label
      className="label"
    >
      Example
    </label>
    <div
      className="control has-icons-right"
    >
      <div
        className="control"
      >
        <input
          className="input is-large"
        />
      </div>
      <span
        className="icon is-warning is-right"
      >
        <i
          className="fas fa-user"
        />
      </span>
      <span />
    </div>
  </div>,
]
`;

exports[`sizes rounded 1`] = `
Array [
  <div
    className="field"
  >
    <label
      className="label"
    >
      Example
    </label>
    <div
      className="control has-icons-right"
    >
      <div
        className="control"
      >
        <input
          className="input is-small is-rounded"
        />
      </div>
      <span
        className="icon  is-right"
      >
        <i />
      </span>
      <span />
    </div>
  </div>,
  <div
    className="field"
  >
    <label
      className="label"
    >
      Example
    </label>
    <div
      className="control has-icons-right"
    >
      <div
        className="control"
      >
        <input
          className="input is-medium is-rounded"
        />
      </div>
      <span
        className="icon  is-right"
      >
        <i />
      </span>
      <span />
    </div>
  </div>,
  <div
    className="field"
  >
    <label
      className="label"
    >
      Example
    </label>
    <div
      className="control has-icons-right"
    >
      <div
        className="control"
      >
        <input
          className="input is-large is-rounded"
        />
      </div>
      <span
        className="icon  is-right"
      >
        <i />
      </span>
      <span />
    </div>
  </div>,
]
`;
